src/data_hub/dataframe.py

```diff
@@ -138,8 +143,12 @@ def __setitem__(self, key, value):
         if self.auto_sync:
             self.to_deeporigin()
         else:
-            # an empty set means "all "
-            self._modified_columns[key] = set()
+            if isinstance(key, str):
+                # an empty set means "all "
+                self._modified_columns[key] = set()
+            elif isinstance(key, list):
+                for item in key:
+                    self._modified_columns[item] = set()
```

Comment:
this fix is for the weird code that the AI assistant wrote where series were inserted into dataframes for every row